lib/src/presentations/main/screen/main_view.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:jobspot/src/core/config/localization/app_local.dart';
+import 'package:jobspot/src/core/constants/constants.dart';
+import 'package:jobspot/src/presentations/main/cubit/main_cubit.dart';
+import 'package:jobspot/src/presentations/main/widgets/custom_bottom_bar.dart';
+import 'package:jobspot/src/presentations/sign_in/widgets/custom_button.dart';
+
+class MainView extends StatelessWidget {
+  const MainView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final cubit = context.read<MainCubit>();
+    return Scaffold(
+      bottomNavigationBar: BlocBuilder<MainCubit, MainState>(
+        buildWhen: (previous, current) =>
+            previous.currentIndex != current.currentIndex,
+        builder: (context, state) {
+          return CustomBottomBar(
+            changeTab: (index) {
+              cubit.changeIndex(index);
+              cubit.toPage(index);
+            },
+            onTap: () => showBottomSheet(context),
+            currentIndex: state.currentIndex,
+          );
+        },
+      ),
+      body: PageView.builder(
+        controller: cubit.pageController,
+        itemCount: 4,
+        onPageChanged: (value) => cubit.changeIndex(value),
+        itemBuilder: (context, index) {
+          return PageStorage(bucket: cubit.bucket, child: cubit.screens[index]);
+        },
+      ),
+    );
+  }
+
+  void showBottomSheet(BuildContext context) {
+    showModalBottomSheet(
+      context: context,
+      shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.vertical(top: Radius.circular(20)),
+      ),
+      backgroundColor: Colors.white,
+      builder: (context) {
+        return Container(
```

Comment:
You should separate widget contain of bottom sheet to new widget class.

Reply:
I fixed it